adbnx_adapter/tests/conftest.py

```diff
@@ -0,0 +1,118 @@
+import os
+import time
+import json
+import requests
+import subprocess
+from pathlib import Path
+from adbnx_adapter.arangoDB_networkx_adapter import ArangoDB_Networkx_Adapter
+
+PROJECT_DIR = Path(__file__).parent.parent.parent
+
+
+def pytest_sessionstart():
+    global con
+    con = get_oasis_crendetials()
+    print_connection_details(con)
+    time.sleep(5)  # Enough for the oasis instance to be ready.
+
+    global adbnx_adapter, imdb_adbnx_adapter, grid_adbnx_adapter, football_adbnx_adapter
+    adbnx_adapter = ArangoDB_Networkx_Adapter(con)
+    imdb_adbnx_adapter = IMDB_ArangoDB_Networkx_Adapter(con)
+    grid_adbnx_adapter = Basic_Grid_ArangoDB_Networkx_Adapter(con)
+    football_adbnx_adapter = Football_ArangoDB_Networkx_Adapter(con)
+
+    arango_restore("examples/data/fraud_dump")
+    arango_restore("examples/data/imdb_dump")
+
+    edge_definitions = [
+        {
+            "edge_collection": "accountHolder",
+            "from_vertex_collections": ["customer"],
+            "to_vertex_collections": ["account"],
+        },
+        {
+            "edge_collection": "transaction",
+            "from_vertex_collections": ["account"],
+            "to_vertex_collections": ["account"],
+        },
+    ]
+    adbnx_adapter.db.create_graph("fraud-detection", edge_definitions=edge_definitions)
+
+
+def pytest_sessionfinish(session, exitstatus):
+    print_connection_details(con)
+
+
+def get_oasis_crendetials() -> dict:
+    url = "https://tutorials.arangodb.cloud:8529/_db/_system/tutorialDB/tutorialDB"
+    request = requests.post(url, data=json.dumps("{}"))
+    if request.status_code != 200:
+        raise Exception("Error retrieving login data.")
+
+    return json.loads(request.text)
+
+
+def arango_restore(path_to_data):
+    restore_prefix = "./" if os.getenv("GITHUB_ACTIONS") else ""  # temporary hack
+
+    subprocess.check_call(
+        f'chmod -R 755 ./arangorestore && {restore_prefix}arangorestore -c none --server.endpoint http+ssl://{con["hostname"]}:{con["port"]} --server.username {con["username"]} --server.database {con["dbName"]} --server.password {con["password"]} --default-replication-factor 3  --input-directory "{PROJECT_DIR}/{path_to_data}"',
+        cwd=f"{PROJECT_DIR}/adbnx_adapter/tests",
+        shell=True,
+    )
+
+
+def print_connection_details(con):
+    print("----------------------------------------")
+    print("https://{}:{}".format(con["hostname"], con["port"]))
+    print("Username: " + con["username"])
+    print("Password: " + con["password"])
+    print("Database: " + con["dbName"])
+    print("----------------------------------------")
+
+
+class IMDB_ArangoDB_Networkx_Adapter(ArangoDB_Networkx_Adapter):
```

Comment:
This is a more general qn (from seeing these subclasses): How often do we expect to see users with multiple NetworkX graphs at the same time? If it isn't that often, disregard this remark.

If it is often, we may want to consider composition instead of inheritance. This could mean splitting out the identification, "keyification", and preparation to separate "behavior" classes so that they can be reused and passed into the `ArangoDB_NetworkX_Adaptor` constructor.

Reply:
sounds interesting... I don't have the answer to your first question but will most likely find someone who does during our ML sync. 

In the meantime I can look into how this would look like in python